SuperChamps/contracts/NFTs/SCAccessPass.sol

```diff
@@ -109,40 +115,52 @@ contract SCAccessPass is ERC721, ISCAccessPass {
     /// @notice Mints an SBT.
     /// @dev Payable function. Requires msg.value to equal price.
     function register() external payable {
-        require(passholderID[_msgSender()] == 0, "ALREADY MINTED SBT");
-        require(msg.value == price, "INVALID PAYMENT");
+        require(passholder_id[_msgSender()] == 0, "ALREADY MINTED SBT");
+        require(msg.value == upgrade_prices[0], "INVALID PAYMENT");
         uint256 tokenId = _tokenIdCounter;
         _safeMint(_msgSender(), tokenId);
-        passholderID[_msgSender()] = tokenId;
+        setPassholderID(_msgSender(), tokenId);
+        pass_level[_msgSender()] = 1;
         _tokenIdCounter++;
     }
 
     /// @notice Mints an SBT to a recipient without requiring payment.
     /// @dev Callable only by Systems Admin.
     /// @param recipient_ The token recipient.
     function freeRegister(address recipient_) external isSystemsAdmin {
-        require(passholderID[recipient_] == 0, "ALREADY MINTED SBT");
+        require(passholder_id[recipient_] == 0, "ALREADY MINTED SBT");
         uint256 tokenId = _tokenIdCounter;
         _safeMint(recipient_, tokenId);
-        passholderID[recipient_] = tokenId;
+        setPassholderID(recipient_, tokenId);
         _tokenIdCounter++;
     }
 
+    /// @notice Mints an SBT.
+    /// @dev Payable function. Requires msg.value to equal price.
+    function upgrade() external payable {
+        require(passholder_id[_msgSender()] > 0, "MUST MINT SBT");
+        uint256 level_ = pass_level[_msgSender()];
+        require(level_ < upgrade_prices.length, "MAX LEVEL");
+        require(msg.value == upgrade_prices[level_], "INVALID PAYMENT");
+        pass_level[_msgSender()] = level_+1;
+    }
+
     /// @notice Burns an SBT from a holder. May be used for future metagame functionality.
     /// @dev Callable only by Systems Admin.
     /// @param holder_ The token holder.
     function burnToken(address holder_) external isSystemsAdmin {
-        uint256 _tokenId = passholderID[holder_];
+        uint256 _tokenId = passholder_id[holder_];
         require(_tokenId != 0, "ALREADY MINTED SBT");
         _burn(_tokenId);
-        passholderID[holder_] = 0;
+        setPassholderID(holder_, 0);
+        pass_level[holder_] = 0;
```

Comment:
Correct. If we were to remove an entry, not only would it be computationally inefficient, it would require re-mapping owner addresses which is a no-go.

Burn will be used very rarely in order to revoke AccessPasses from users who for whatever reason need to be banned from the service.